File: differ/src/main/java/ru/starksoft/differ/utils/hash/Cache.java
package ru.starksoft.differ.utils.hash;

import java.util.Map;
import java.util.concurrent.ConcurrentHashMap;

import androidx.annotation.NonNull;
import androidx.annotation.VisibleForTesting;

public final class Cache {

	private final Map<String, Integer> map = new ConcurrentHashMap<>();

	public final int getValue(@NonNull String key) {
		Integer value = map.get(key);

		if (value == null) {
			value = key.hashCode();
			map.put(key, value);
		}

		return value;
	}


	@VisibleForTesting
	public int getCacheSize() {
		return map.size();
	}
}

File: diff-adapter/src/main/java/ru/starksoft/differ/utils/hash/Cache.kt
package ru.starksoft.differ.utils.hash

import androidx.annotation.VisibleForTesting
import java.util.concurrent.ConcurrentHashMap

class Cache {

    private val map = ConcurrentHashMap<String, Int>()

    val cacheSize: Int
        @VisibleForTesting
        get() = map.size

    fun getValue(key: String): Int {
        var value = map[key]

        if (value == null) {
            value = key.hashCode()
            map[key] = value
        }

        return value
    }
}
